### PR TITLE
test(fiatAmount): test UsdPricesUpdater

### PR DESCRIPTION
**Pull request by shoom3301**

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/usdAmount/updaters/UsdPricesUpdater.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ const swrOptions: SWRConfiguration = {
   revalidateOnFocus: true,
 }
 
-export function UsdPricesUpdater() {
+export function UsdPricesUpdater({ id }: { id?: string }) {
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/usdAmount/updaters/UsdPricesUpdater.tsx:32]: I imagine the answer is no, but is there any way to fool/mock swr without adding this test-exclusive parameter?
That doesn't feel right.

In the worst case, make it clear this is used only for testing.

Maybe

```suggestion
export function UsdPricesUpdater({ testCacheIdSuffix }: { testCacheIdSuffix = '' }) {
```

**Review by alfetopito — COMMENTED** [review]
Great tests!

Some comments.

Also, there's 1 test failing :(

File: apps/cowswap-frontend/src/modules/usdAmount/updaters/UsdPricesUpdater.test.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      return (
+        <JotaiTestProvider store={store} initialValues={initialValues}>
+          {children}
+        </JotaiTestProvider>
+      )
+    },
+  }
+}
+
+async function setupTest(
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/usdAmount/updaters/UsdPricesUpdater.test.tsx:52]: performTest sounds better, thanks!

File: apps/cowswap-frontend/src/modules/usdAmount/updaters/UsdPricesUpdater.test.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+  await waitFor(() => resolvesCount === waitForResolvesCount)
+
+  return store.get(usdRawPricesAtom)
+}
+
+describe('UsdPricesUpdater', () => {
+  afterEach(() => {
+    jest.clearAllMocks()
+    jest.resetAllMocks()
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/usdAmount/updaters/UsdPricesUpdater.test.tsx:80]: `resetAllMocks` is actually enough, thanks!

File: apps/cowswap-frontend/src/legacy/state/swap/utils.ts [hunk 4]
```diff
@@ -20,44 +24,69 @@ export function isWrappingTrade(
...
-      =========
-      Input Amount [w/FEE]: ${trade.inputAmountWithFee.toExact()}
-      Output Amount:        ${trade.outputAmount.toExact()}
-      =========
-      Fee Amount [as SELL]: ${trade.fee?.feeAsCurrency?.toExact()} ${trade.inputAmount.currency.symbol}
-      =========
-      Maximum Sold:         ${trade.fee?.feeAsCurrency && trade.maximumAmountIn(allowedSlippage).toExact()}
-    `
-    )
+function parseIndependentFieldURLParameter(urlParam: any): Field {
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/legacy/state/swap/utils.ts:27]: Just moved functions from `hooks.tsx` using auto refactoring.

Comment (shoom3301) [reply apps/cowswap-frontend/src/legacy/state/swap/utils.ts:27]: To fix circular dependency problem

File: apps/cowswap-frontend/src/legacy/state/swap/reducer.ts [hunk 5]
```diff
@@ -15,11 +15,12 @@ import {
   switchCurrencies,
   typeInput,
 } from 'legacy/state/swap/actions'
-import { queryParametersToSwapState } from 'legacy/state/swap/hooks'
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/legacy/state/swap/reducer.ts:18]: Here I broke Circular dependency.
`reducer` and `hooks` were in a circle

File: apps/cowswap-frontend/src/legacy/state/swap/utils.ts (hunk 4 above)
Comment (alfetopito) [reply apps/cowswap-frontend/src/legacy/state/swap/utils.ts:27]: What do you mean by `auto refactoring`?

**Review by alfetopito — APPROVED** [review]

File: apps/cowswap-frontend/src/legacy/state/swap/utils.ts (hunk 4 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/legacy/state/swap/utils.ts:27]: image

**Pull request merged**